backend/productsearch/db/redis_helpers.py

```diff
@@ -32,5 +28,5 @@ def get_test_index():
 async def get_async_index():
     global global_index
     if not global_index:
-        global_index = AsyncSearchIndex(schema, client)
+        global_index = AsyncSearchIndex(schema, redis_url=config.REDIS_URL)
```

Comment:
- share the same redis client and connection pool (global instance)
- create async search index using that (global instance)

lifecycle hook to clean it up

Reply:
Don't enter into context window through FastAPI dependency